tests/test_transforms/test_math/test_limit_transform.py

```diff
@@ -0,0 +1,129 @@
+):
+    """Check forward scaled logit transform works correctly on nontarget column"""
+    preprocess = LimitTransform(in_column="non_target", lower_bound=lower_bound, upper_bound=upper_bound)
+    result = preprocess.fit_transform(ts=ts_check_limit_transform).to_pandas()
+    np.testing.assert_array_almost_equal(
+        result["segment_1"]["non_target"], result["segment_1"]["non_target_transformed_" + n_test]
+    )
+
+
+@pytest.mark.parametrize("lower_bound,upper_bound", [(-1e10, 1e10), (-15, 100)])
+def test_inverse_transform(ts_check_limit_transform: TSDataset, lower_bound: float, upper_bound: float):
+    """Check that inverse_transform rolls back transform result"""
+    preprocess = LimitTransform(in_column="target", lower_bound=lower_bound, upper_bound=upper_bound)
+    preprocess.fit_transform(ts=ts_check_limit_transform)
+    preprocess.inverse_transform(ts=ts_check_limit_transform)
+    result = ts_check_limit_transform.to_pandas()
+    np.testing.assert_array_almost_equal(result["segment_1"]["target"], result["segment_1"]["target_no_change"])
+
+
+@pytest.mark.parametrize("lower_bound,upper_bound", [(-10, 200), (-30, 50), (-5, 5)])
+def test_fit_transform_values_out_of_borders(
+    ts_check_pipeline_with_limit_transform: TSDataset, lower_bound: float, upper_bound: float
+):
+    """Check that Exception raises when there are values out of bounds"""
+    transform = LimitTransform(in_column="target", lower_bound=0, upper_bound=10)
+    with pytest.raises(ValueError):
+        transform.fit_transform(ts_check_pipeline_with_limit_transform)
+
+
+def test_pipeline_with_limit_transform(ts_check_pipeline_with_limit_transform: TSDataset):
+    """Check that forecasted target and quantiles are in given range with LimitTransform"""
+    model = ProphetModel()
+    transform = LimitTransform(in_column="target", lower_bound=-20, upper_bound=105)
+    pipeline = Pipeline(model=model, transforms=[transform], horizon=10)
+    pipeline.fit(ts_check_pipeline_with_limit_transform)
+    forecast_ts = pipeline.forecast(prediction_interval=True)
+    forecast_df = forecast_ts.to_pandas()
+    features = forecast_df.loc[:, pd.IndexSlice[:, ["target", "target_0.025", "target_0.975"]]]
+    assert (features >= -20).all().all() and (features <= 105).all().all()
+
+
+def test_pipeline_without_limit_transform(ts_check_pipeline_with_limit_transform: TSDataset):
+    """Check that forecasted target is not in given range without LimitTransform"""
+    model = ProphetModel()
+    pipeline = Pipeline(model=model, horizon=10)
+    pipeline.fit(ts_check_pipeline_with_limit_transform)
+    forecast_ts = pipeline.forecast()
+    forecast_df = forecast_ts.to_pandas()
+    assert (forecast_df < -20).any().any() or (forecast_df > 105).any().any()
+
+
+def test_backtest(ts_check_pipeline_with_limit_transform: TSDataset):
+    """Check that backtest function executes without errors"""
+    model = ProphetModel()
```

Comment:
Better to use more lightweight model like `NaiveModel`(tests are already works slowly)